examples/embeddable_examples/public/react_embeddables/search/search_react_embeddable.tsx

```diff
@@ -38,67 +37,82 @@ export const getSearchEmbeddableFactory = (services: Services) => {
 
       const api = buildApi(
         {
-          ...timeRangeApi,
+          ...timeRange.api,
           dataViews: dataViews$,
           dataLoading: dataLoading$,
           serializeState: () => {
             return {
               rawState: {
-                ...serializeTimeRange(),
+                ...timeRange.serialize(),
               },
               references: [],
             };
           },
         },
         {
-          ...timeRangeComparators,
+          ...timeRange.comparators,
         }
       );
 
+      let isUnmounted = false;
+      const error$ = new BehaviorSubject<Error | undefined>(undefined);
+      const count$ = new BehaviorSubject<number>(0);
+      const onFetch = (fetchContext: FetchContext, isCanceled: () => boolean) => {
+        error$.next(undefined);
+        if (!defaultDataView) {
+          return;
+        }
+        dataLoading$.next(true);
+        getCount(
+          defaultDataView,
+          services.data,
+          fetchContext.filters ?? [],
+          fetchContext.query,
+          // timeRange and timeslice provided seperatly so consumers can decide
+          // whether to refetch data for just mask current data.
+          // In this example, we must refetch because we need a count within the time range.
+          fetchContext.timeslice
+            ? {
+                from: new Date(fetchContext.timeslice[0]).toISOString(),
+                to: new Date(fetchContext.timeslice[1]).toISOString(),
+                mode: 'absolute' as 'absolute',
+              }
+            : fetchContext.timeRange
+        )
+          .then((nextCount: number) => {
+            if (isUnmounted || isCanceled()) {
+              return;
+            }
+            dataLoading$.next(false);
+            count$.next(nextCount);
+          })
+          .catch((err) => {
+            if (isUnmounted || isCanceled()) {
+              return;
+            }
+            dataLoading$.next(false);
+            error$.next(err);
+          });
+      };
+      const unsubscribeFromFetch = onFetchContextChanged({
+        api,
+        onFetch,
+        fetchOnSetup: true,
+      });
+
       return {
         api,
         Component: () => {
-          const [count, setCount] = useState<number>(0);
-          const [error, setError] = useState<Error | undefined>();
-          const [filters, query, appliedTimeRange] = useBatchedPublishingSubjects(
-            api.parentApi?.filters$,
-            api.parentApi?.query$,
-            appliedTimeRange$
-          );
+          const count = useStateFromPublishingSubject(count$);
```

Comment:
nit: you could maybe batch these? I think we removed any render downsides to that. If so, maybe it should be our best practice